src/settingsview.cpp

```diff
@@ -176,27 +180,28 @@ void SettingsView::setReopenTab(bool reopen)
 
 void SettingsView::onDownloadDirChanged(const QString &dir)
 {
-    ui->downloadDirPath->setText(formatDownloadDir(dir));
+    ui->downloadDirPath->setText(formatSettingsDir(dir));
     ui->downloadDirPath->setToolTip(dir);
 }
 
-void SettingsView::copyDownloadPathToClipboard()
+void SettingsView::copySettingsPathToClipboard(QPushButton* button)
 {
-    QString downloadPath = KiwixApp::instance()->getSettingsManager()->getDownloadDir();
-    QApplication::clipboard()->setText(downloadPath);
+    auto settingsManager = KiwixApp::instance()->getSettingsManager();
+    QString pathToCopy = (button == ui->downloadDirPathCopy) 
+        ? settingsManager->getDownloadDir() 
+        : settingsManager->getMonitorDir();
```

Comment:
This (or the function signature and semantics) is a little ugly. Either pass `pathToCopy` as an argument as well, or change the parameter type to something else (e.g. an enum or a string) based on which select both `button` and `pathToCopy` inside the function body.

Reply:
Yeah, did not really like it as well but was not sure which was the best option. I am going to pass it as an argument to provide modularity also for future use.